### PR TITLE
fix: update path logic to support running on Windows

### LINKED ISSUE #32: Windows issues with filepath strings
I gave it a quick try on Windows. I found a few issues.

https://github.com/influxdata/pkg-config/blob/b10a85ae06c3d25a01f905ba42d0544c40933824/main.go#L37

This will contain "C:\\" (or another letter) on Windows.

https://github.com/influxdata/pkg-config/blob/b10a85ae06c3d25a01f905ba42d0544c40933824/libs/flux/build.go#L394

This will contain something like C:\\installpath\\github.com\\influxdata\\flux@v0.104.0

https://github.com/influxdata/pkg-config/blob/b10a85ae06c3d25a01f905ba42d0544c40933824/internal/modload/init.go#L52

This will contain "C:\\" on Windows. This last one causes an infinite loop.

### PR DESCRIPTION
**Pull request by danxmoran**
Closes #32

EDIT: This + influxdata/flux#3874 got Windows builds of `influxdb` working for me

### CODE REVIEW

File: internal/modload/init.go [hunk 1]
```diff
@@ -49,7 +49,9 @@ func HasModRoot() bool {
 // of a vendor directory.
 func findVendorMod(cwd string) (string, bool) {
 	cwd = filepath.Dir(cwd)
-	for cwd != "/" {
+	// According to its godoc, the return value for `filepath.Dir` will
+	// only end in a path separator when it's the root directory.
```
Comment (rockstar) [inline internal/modload/init.go:53]: This is a great comment.

File: libs/flux/constants_windows.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+// fluxVersionRegexp is used to extract the version of flux pulled down by `go mod` by inspecting
+// its path on the filesystem.
+var fluxVersionRegexp = regexp.MustCompile(`\\github\.com\\influxdata\\flux@(v\d+\.\d+\.\d+.*)$`)
+
+// pcSep is the separator used between components in all the path-fields written into `flux.pc`
+// by our `pkg-config` wrapper. On Windows we have to double-escape the OS's path separator because:
+//   1. The 1st escape is "used" when `pkg-config` prints the paths to the caller (some piece of `go build`)
+//   2. The 2nd escape is "used" when `go build` passes the paths to `ld`
+// With only a single-escape of the path separator, `go build` ends up passing args like `-Lmypathtoflux`
+// instead of `-Lmy\path\to\flux`, and linking fails.
```
Comment (rockstar) [inline libs/flux/constants_windows.go:14]: I read this and now I'm annoyed. 😂

**Review by rockstar — APPROVED** [review]
Thanks for doing this. I don't think we have anyone on the team using windows. Were all windows builds just outright failing? That seems like something we should chase, at least getting it into CI.

**Comment by danxmoran** [comment]
> Were all windows builds just outright failing?

Cross-builds from Linux->Windows worked OK since all this tooling runs at build-time, but from the errors I saw I'd expect native Windows builds have never worked. TBH I don't even know if _all_ native Windows builds work with these changes, but it gets things working in CircleCI's MinGW + bash env

**Pull request merged**